Lib/unittest/mock.py

```diff
@@ -648,9 +648,18 @@ def reset_mock(self,  visited=None,*, return_value=False, side_effect=False):
         if side_effect:
             self._mock_side_effect = None
 
-        for child in self._mock_children.values():
+        for key, child in self._mock_children.items():
             if isinstance(child, _SpecState) or child is _deleted:
                 continue
+            if (
+                return_value
```

Comment:
What happens here, for, say a `MagicMock` with `__iter__` mocked out and returning an empty list or empty tuple? Could you add a test showing that?

(My concern is that these "false" value incorrectly end up resulting in the `return_value` being set to `False`.

Reply:
Done!